### PR TITLE
Enable lazy ClangImporter diagnostics by default

### PR DESCRIPTION
**Pull request by NuriAmari**
Replace the existing `-enable-experimental-clang-importer-diagnostics`
flag with an opt-out version entitled `-disable-experimentalc-clang-importer-diagnostics`.
Enable the beviour previously hidden behind the old flag by default.

Implements the changes described in [this forum post](https://forums.swift.org/t/pitch-lazy-clangimporter-diagnostics-enabled-by-default/54651).

### CODE REVIEW

**Review by zoecarver — APPROVED** [review]
This looks good to me.

However, before this lands, I want to make sure there is at least one test where there are multiple un-importable APIs and we only use _one_ of them. We should only get one error.

File: test/ClangImporter/experimental_diagnostics_no_noise.swift
Comment (zoecarver) [inline test/ClangImporter/experimental_diagnostics_no_noise.swift:62]: Not needed.

Comment (zoecarver) [reply test/ClangImporter/experimental_diagnostics_no_noise.swift:62]: Just for future reference (you already check-not this above). This is fine for now, though. I want to land this 😁

**Comment by zoecarver** [comment]
@swift-ci please smoke test.

**Comment by zoecarver** [comment]
As discussed offline, I just realized because these tests require objc-interop they are only getting run on macOS. Please add a test that uses C++ interop so they get run on all platforms. You can do something like this:
```
// RUN: %target-typecheck-verify-swift -I %S/Inputs -enable-cxx-interop
```

**Comment by zoecarver** [comment]
@swift-ci please smoke test

**Comment by zoecarver** [comment]
@swift-ci please smoke test

**Comment by zoecarver** [comment]
@swift-ci please test Windows

File: test/ClangImporter/attr-swift_name.swift
Comment (zoecarver) [inline test/ClangImporter/attr-swift_name.swift:2]: Why did this test need to be updated?

**Comment by zoecarver** [comment]
Don't merge this until we resolve the updated test. I also want to understand why these can't be errors anymore.

**Comment by NuriAmari** [comment]
@swift-ci please smoke test

File: test/ClangImporter/experimental_diagnostics_opt_out.swift
Comment (NuriAmari) [inline test/ClangImporter/experimental_diagnostics_opt_out.swift:19]: TODO: Remove unnecessary CHECK-NOTs

**Comment by zoecarver** [comment]
@swift-ci please smoke test.

File: test/ClangImporter/attr-swift_name.swift
Comment (NuriAmari) [reply test/ClangImporter/attr-swift_name.swift]: It doesn't need updating anymore, but at the time it was because the new diagnostics were tripping a CHECK-NOT. The diagnostic was accurate, and is still there, but now it's emmitted later in a position that does not trip a CHECK-NOT.

The new diagnostic informs about a circularFriends method that can't be imported because parameter type `CircularName` is unimportable. Not exactly sure of the details, but looks intentional, I can dig deeper if need be.

**Comment by NuriAmari** [comment]
MacOS failure looks unrelated to me. I will complete the TODO comment I have left over above, and will test with errors instead of notes here: https://github.com/apple/swift/pull/41162. If that goes well, should be ready for merge.

**Comment by zoecarver** [comment]
@swift-ci please smoke test macOS

**Review by zoecarver — APPROVED** [review]

**Pull request merged**
